cli/internal/config/config_file_test.go

```diff
@@ -9,6 +9,51 @@ import (
 	"github.com/vercel/turborepo/cli/internal/fs"
 )
 
+func Test_UserConfigPath(t *testing.T) {
+	fsys := afero.NewMemMapFs()
+
+	tests := []struct {
+		name string
+		fsys afero.Fs
+	}{
+		{
+			name: "get doesn't create a path",
+			fsys: fsys,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// XDG is not filesystem aware. Clean up first.
+			path, _ := createUserConfigPath(fsys)
+			err := os.Remove(path.Dir().ToString())
+
+			if err != nil {
+				t.Errorf("failed to clean up first: %v", err)
+			}
+
+			gotConfigPath, getErr := getUserConfigPath(tt.fsys)
+			if gotConfigPath == "" || getErr == nil {
```

Comment:
What is this doing? If we've failed to get the path `gotConfigPath == ""` or we've succeeded `getErr == nil`, we do the assignment? 

Presumably we should know which case we want and this should be an assertion instead?

Reply:
This was the consequence of a boolean algebra reduction, I'll make it more clear what is going on here.

Reply:
Good catch! This was *erroneous* boolean algebra reduction.